### ISSUE TITLE
Prevent quotes based on internal buffer trading

### ISSUE THREAD

**Issue by MartinquaXD** [cowprotocol/services#950]
Our trade verification logic is supposed to make sure that quoted prices are actually realistic and achievable by simulating the trade route (call_data) returned with the quote.

However, the current implementation doesn't make sure that the quoted price wasn't achieved by internal buffer trading. With internal buffer trading external solvers could generate trade routes that simulate and give unrealistically good prices.

Note that this loophole only becomes a real issue when we allow parties with knowledge about our settlement contract balances to generate quotes. For example when we get quotes from solvers once the solver-driver colocation is done.

Preventing this issue is not trivial so let's discuss our options.

**Comment by fleupold** [comment]
Closing in favor of #1662
